common/src/main/java/com/wynntils/features/user/GuildStarsToRankFeature.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.wynn.event.ChatMessageReceivedEvent;
+import com.wynntils.wynn.objects.GuildRank;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.HoverEvent;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class GuildStarsToRankFeature extends UserFeature {
+
+    private static final Pattern guildMessage = Pattern.compile("§3\\[§b(★*)§3(.+)§3\\] (.*)");
+
+    @SubscribeEvent
+    public void onChat(ChatMessageReceivedEvent event) {
+        String message = event.getMessage().getString();
+        Matcher match = guildMessage.matcher(message);
+        if (!userEnabled || !(match.matches())) return;
+        MutableComponent newMessage = new TextComponent(String.format(
+                "§3[§b%s §3%s§3] §b", GuildRank.fromInt(match.group(1).length()).getRank(), match.group(2)));
+
+        MutableComponent tc = new TextComponent("");
+        String sentMessage = match.group(3);
+        List<String> urls = extractUrls(sentMessage);
+        int previousIndex = 0;
+        if (urls.isEmpty()) {
+            event.setMessage(newMessage.append(tc.append(sentMessage)));
+            return;
+        }
+        for (String url : urls) {
+            int index = sentMessage.indexOf(url);
+            tc.append(sentMessage.substring(previousIndex, index));
+            TextComponent urlText = new TextComponent(url);
+            ClickEvent clickEvent = new ClickEvent(ClickEvent.Action.OPEN_URL, url);
+            HoverEvent hoverEvent = new HoverEvent(HoverEvent.Action.SHOW_TEXT, new TextComponent("Open " + url));
+            tc.append(urlText.setStyle(
+                    urlText.getStyle().withClickEvent(clickEvent).withHoverEvent(hoverEvent)));
+            previousIndex = index + 1;
+        }
+
+        event.setMessage(newMessage.append(tc));
+    }
+
+    private static List<String> extractUrls(String text) {
+        List<String> containedUrls = new ArrayList<String>();
+        String urlRegex = "((https?|ftp|gopher|telnet|file|http):((//)|(\\\\))+[\\w\\d:#@%/;$()~_?\\+-=\\\\\\.&]*)";
```

Comment:
`gopher`? :-D

But seriously, what is this even for?

Reply:
This seems to be an unrelated feature for adding clickevent to urls

Reply:
Its not unrelated as before whwn this feature was enabled, clicking on url in guold chat disabled. This is the fix unless anyone else knows better

Reply:
Maybe fix the underlying issue instead of making a jank fix?

Reply:
how else would i do this? the issue comes from setting the message from the one with a clickevent to one without. i am simply adding the clickevent. how do you propose i do this?

Reply:
> `gopher`? :-D
> 
> But seriously, what is this even for?

also gopher is another protocol like https idk if u alr knew that

Reply:
Well the problem is that we reconstruct the component isn't it? Then maybe try not reconstructing it?

Reply:
Is it even possible to edit a component?

Reply:
you can copy component contents/style while making whatever changes necessary, instead of trying to reconstruct it from the ground up. code in `ChatItemModel` achieves a similar goal to this feature while preserving the rest of the message